### DISCUSSION TITLE
Callback function for adding a stroke through plugin

### DISCUSSION THREAD

**Discussion by MiltonBalaOfficial** [xournalpp/xournalpp#5978] [Q&A] [answered]
Actually, I want to make a plugin for inserting some basic shapes (different types of triangles, arrow with different size arrowhead, line with arrowhead on the middle etc.) quickly through shortcut. But don't know about stroke table format. So, if anyone give me the code for the callback function, I may be able to make a very handy plugin "Quick Shape". Thanks in advance.

**Comment by rolandlo** [comment, marked as answer]
An example for adding a triangle:
```lua
triangle = { 
  x = { 300.0, 500.0, 400.0 }, 
  y = { 300.0, 300.0, 450.0 }, 
  width = 1.5, 
  tool = "pen", 
  color = 0xff0000,
}
app.addStrokes { strokes = { triangle } }
app.refreshPage()
```

Note that when working with the Lua language server using the file `luapi_application.def.lua` you get some help on hover while coding.
![Bildschirmfoto vom 2024-10-23 07-02-09](https://github.com/user-attachments/assets/ea5a33c3-fd95-41c3-b307-2e9b829f9419)

**Reply by MiltonBalaOfficial** [reply]
Thanks a lot. Though my coding skill tends to zero, with your help now I am able to insert different shapes just by pressing shortcut keys. Its very helpful to have shapes like this:
![Screenshot 2024-10-23 113405](https://github.com/user-attachments/assets/0400c24c-e8ca-49e3-a1f8-e8fc257c2bf5)
